Headers/DebugServer2/Target/Windows/Process.h

```diff
@@ -52,6 +52,16 @@ class Process : public ds2::Target::ProcessBase,
   ErrorCode writeMemory(Address const &address, void const *data, size_t length,
                         size_t *nwritten = nullptr) override;
 
+private:
+  ErrorCode makeMemoryWritable(Address const &address, size_t length,
+                               std::vector<MemoryRegionInfo> &modifiedRegions);
+  ErrorCode
+  restoreRegionsProtection(const std::vector<MemoryRegionInfo> &regions);
+  ErrorCode getMemoryRegionInfoInternal(Address const &address,
+                                        MemoryRegionInfo &info);
+  DWORD convertMemoryProtectionFromLLDB(uint32_t protection);
```

Comment:
These are not converting to/from LLDB memory protection types, they're converting from ds2 to windows type and vice versa, so you should just call them to/from Windows I think. There are other functions like these for file mode conversions in one of @fjricci's patches. Look at how he did that maybe?